modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/config/TerraformBootConfig.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.deployment.deployers.terraform.config;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import lombok.Getter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+/**
+ * Configuration class of terraform-boot env.
+ */
+@Configuration
+@Profile("terraform-boot")
+public class TerraformBootConfig {
+
+    @Getter
+    @Value("${server.port}")
+    private String clientPort;
+
+    @Getter
+    @Value("${webhook.request.base.uri}")
+    private String clientBaseUri;
+
+    @Getter
+    @Value("${webhook.deployCallbackUri}")
+    private String deployCallbackUri;
+
+    @Getter
+    @Value("${webhook.destroyCallbackUri}")
+    private String destroyCallbackUri;
+
+    @Value("${terraform.boot.terraform.script.module.directory}")
+    private String moduleParentDirectoryPath;
+
+    /**
+     * Get moduleParentDirectoryPath.
+     */
+    public String getModuleParentDirectoryPath() {
+        if (moduleParentDirectoryPath.isBlank() || moduleParentDirectoryPath.isEmpty()) {
+            this.moduleParentDirectoryPath =
+                    System.getProperty("java.io.tmpdir");
+        } else {
+            this.moduleParentDirectoryPath = moduleParentDirectoryPath;
+        }
+        return moduleParentDirectoryPath;
+    }
+
+    /**
+     * Get tBase Client Request Url.
+     */
+    public String getClientRequestBaseUrl() throws UnknownHostException {
```

Comment:
we must make this also configurable. 
Only if no configuration is available, we must use the http://localhost:${server.port}.
Configuration is needed, in case we are using docker network or using load balancers in between.